### ISSUE TITLE
SetContentMD5 not being set for PUT requests

### ISSUE THREAD

**Issue by grantlouisherman** [aws/aws-sdk-cpp#2818]
### Describe the bug

When I try to SetContentMD5 using the S3 api for c++, the value never gets set. I have used the client directly and I am able to set the md5 content, so it must be a bug in the c++ implementation. There is a discussion open here: https://github.com/aws/aws-sdk-cpp/discussions/2433 that describes the same issue, but it seems like this hasnt gotten any attention for a bit.  There are a few issues open on Stackoverflow, https://stackoverflow.com/questions/75985503/why-s3-setcontentmd5-is-no-use-in-my-program, where the same issue is brought up.

This is a super critical API for my team so it would be great if we could get some movement on this issue. 

### Expected Behavior

I am able to SetContentMD5 in a PUT request

### Current Behavior

The value in SetContentMD5 is ignored

### Reproduction Steps

```cpp
// Per the StackOverflow example listed above because I cant share company code

void put_object_buf(Aws::String& bucketName, Aws::String& objectName, S3::S3Client& s3_client)
{
    Aws::S3::Model::PutObjectRequest putObjectRequest;
    putObjectRequest.SetBucket(bucketName);

    std::shared_ptr<Aws::IOStream> bigStream = Create5MbStreamForUploadPart("La");

    putObjectRequest.SetBody(bigStream);
    putObjectRequest.SetContentLength(static_cast<long>(putObjectRequest.GetBody()->tellp()));
//    putObjectRequest.SetContentMD5(Aws::Utils::HashingUtils::Base64Encode(Aws::Utils::HashingUtils::CalculateMD5(*putObjectRequest.GetBody())));
    putObjectRequest.SetContentMD5("123");  // still put success
    putObjectRequest.SetKey(objectName);

    Aws::S3::Model::PutObjectOutcome outcome = s3_client.PutObject(putObjectRequest);

    if (!outcome.IsSuccess()) {
        std::cerr << "Error: PutObjectBuffer: " <<
                  outcome.GetError().GetMessage() << std::endl;
    }
    else {
        std::cout << "Success: Object '" << objectName << "' uploaded to bucket '" << bucketName << "'." << std::endl;
        std::cout << "outcome crc32: " << outcome.GetResult().GetChecksumCRC32() << std::endl;
        std::cout << "outcome etag: " << outcome.GetResult().GetETag() << std::endl;
    }
}
```

### AWS CPP SDK version used

current

### Compiler and Version used

n/a

### Operating System and version

mac

**Comment by jmklix** [comment]
Can you specify the exact version of the sdk that you are using? You said `current` but there have been some changes recently regarding `SetContentMD5`.

**Comment by grantlouisherman** [comment]
@jmklix  we are using  C++17.

**Comment by jmklix** [comment]
Can you specify the version of the sdk, not the version of C++ that you are using?

**Comment by sbiscigl** [comment]
> 1.33.7 AFAIK

thats not a version of the C++ SDK were are you getting that number from?

[Also the fix is out in PR](https://github.com/aws/aws-sdk-cpp/pull/2826)

**Comment by sbiscigl** [comment]
was released today with tag 1.11.253. resolving
